src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/JsonSourceGenerator.cs

```diff
@@ -16,47 +19,175 @@ namespace System.Text.Json.SourceGeneration
 using System;
+using System.Collections.Generic;
+
 namespace HelloWorldGenerated
-{
-    public static class HelloWorld
-    {
-        public static string SayHello() 
-        {
-            return ""Hello"";
-");
+{{
+    public class {entry.Key}ClassInfo
+    {{
+        public {entry.Key}ClassInfo() {{ }}
 
-            sourceBuilder.Append(@"
-        }
-    }
-}");
+        private List<string> ClassCtors = new List<string>()
+        {{ {foundCtors} }};
+        private Dictionary<string, string> ClassCtorParams = new Dictionary<string, string>()
+        {{ {foundCtorParams} }};
+        private List<string> ClassMethods = new List<string>()
+        {{ {foundMethods} }};
+        private Dictionary<string, string> ClassFields = new Dictionary<string, string>()
+        {{ {foundFields} }};
+        private Dictionary<string, string> ClassProperties = new Dictionary<string, string>()
+        {{ {foundProperties} }};
+
+        public string GetClassName()
+        {{
+            return ""{entry.Key}ClassInfo"";
+        }}
+
+        public List<string> Ctors
+        {{ get {{ return ClassCtors; }} }}
+
+        public Dictionary<string, string> CtorParams
+        {{ get {{ return ClassCtorParams; }} }}
+
+        public List<string> Methods 
+        {{ get {{ return ClassMethods; }} }}
 
-            context.AddSource("helloWorldGenerated", SourceText.From(sourceBuilder.ToString(), Encoding.UTF8));
+        public Dictionary<string, string> Fields
+        {{ get {{ return ClassFields; }} }}
+
+        public Dictionary<string, string> Properties
+        {{ get {{ return ClassProperties; }} }}
+    }}
+}}
+", Encoding.UTF8));
+            }
         }
 
         public void Initialize(InitializationContext context)
         {
             context.RegisterForSyntaxNotifications(() => new JsonSerializableSyntaxReceiver());
         }
 
-        // Temporary function for now that reads all types. Should search types with attribute JsonSerializable.
-        internal class JsonSerializableSyntaxReceiver : ISyntaxReceiver
+        public class JsonSerializableSyntaxReceiver : ISyntaxReceiver
         {
-            public List<TypeDeclarationSyntax> GeneratorInputTypes = new List<TypeDeclarationSyntax>();
+            public List<KeyValuePair<string, IdentifierNameSyntax>> ExternalClassTypes = new List<KeyValuePair<string, IdentifierNameSyntax>>();
+            public List<KeyValuePair<string, TypeDeclarationSyntax>> InternalClassTypes = new List<KeyValuePair<string, TypeDeclarationSyntax>>();
 
             public void OnVisitSyntaxNode(SyntaxNode syntaxNode)
             {
-                // Get all the type decl in all syntax tree.
-                if (syntaxNode is TypeDeclarationSyntax tds)
+                // Look for classes or structs for JsonSerializable Attribute.
+                if (syntaxNode is ClassDeclarationSyntax || syntaxNode is StructDeclarationSyntax)
```

Comment:
If multiple assemblies have types to have code generated for (indicated with the new attribute), do all of them show up here or is it just the ones in the assembly that fired off the generation?

Reply:
Only attributes from the current assembly trigger the codegen.